service/grpc/onboardDevice.go

```diff
@@ -160,18 +192,20 @@ func onboardInsecureDevice(ctx context.Context, dev *device, links schema.Resour
 	case req.GetCoapGatewayAddress() == "":
 		return fmt.Errorf("invalid URL")
 	}
-	var link schema.ResourceLink
-
-	for _, l := range links {
-		if strings.SliceContains(l.ResourceTypes, cloud.ResourceType) {
-			link = l
-			break
-		}
-	}
-	if link.Href == "" {
+	cloudLinks := links.GetResourceLinks(cloud.ResourceType)
+	if len(links) == 0 {
```

Comment:
Is this correct, you get `cloudLinks` but check the length of `links`? Is `cloudLinks` guaranteed to have at least one element, otherwise `cloudLinks[0]` on line 199 might be out of bounds

Reply:
fixed